data-raw/remaining_carbon_budgets.R

```diff
@@ -1,15 +1,55 @@
-        "GECO2021",       "coal",                 84934.79, "Mt CO2 (2019)",              0.25,
-        "GECO2021",        "oil",                 82987.79, "Mt CO2 (2019)",              0.24,
-        "GECO2021",        "gas",                 49094.21, "Mt CO2 (2019)",              0.14,
-        "GECO2021",      "power",                 91121.17, "Mt CO2 (2019)",              0.27,
-        "GECO2021",      "steel",                 10746.03, "Mt CO2 (2019)",              0.03,
-        "GECO2021", "automotive",                  11338.9, "Mt CO2 (2019)",              0.03,
-        "GECO2021",        "hdv",                  7388.01, "Mt CO2 (2019)",              0.02,
-        "GECO2021",   "aviation",                  5963.32, "Mt CO2 (2019)",              0.02
+# The Extended file data for WEO 2023 is stored here:
+# https://portal.azure.com/#view/Microsoft_Azure_FileStorage/FileShareMenuBlade/~/browse/storageAccountId/%2Fsubscriptions%2Ffeef729b-4584-44af-a0f9-4827075512f9%2FresourceGroups%2FRMI-SP-PACTA-PROD%2Fproviders%2FMicrosoft.Storage%2FstorageAccounts%2Fpactarawdata/path/scenario-sources/protocol/SMB
+# The specific file is 'weo_2023-20240222/WEO2023 extended data/WEO2023_Extended_Data.xlsx'
+# Please download the file and replace the dummy path in variable 'file_scenario_emissions_data'
+# to the path on your computer before running the code.
+file_scenario_emissions_data <- "PATH/TO/WEO2023/SCENARIO/EXTENDED/DATA"
+carbon_emissions <- read_xlsx(file_scenario_emissions_data,
+                              sheet = "World CO2 Emissions",
+                              range = "J7:AP45") %>%
+  select(-c("2022...5", "2030...6", "2035...7", "2040...8", "2045...9", "2050...10", "...11", "2030...12", "2050...13",
+            "2022...15", "2030...16", "2035...17", "2040...18", "2045...19", "2050...20", "...21", "2030...22", "2050...23", "...24",
+            "...31", "2030...32", "2050...33", "...14")) %>%
+  rename(
+    sector = "...1",
+    emissions_2022 = "2022...25",
+    emissions_2030 = "2030...26",
+    emissions_2035 = "2035...27",
+    emissions_2040 = "2040...28",
+    emissions_2045 = "2045...29",
+    emissions_2050 = "2050...30",
+  ) %>%
+  filter(sector %in% c("Coal", "Oil", "Natural gas", "Electricity and heat sectors", "Iron and steel**", "Cement**", "Passenger cars", "Aviation")) %>%
+  mutate(
+    scenario_source = "WEO2023",
+    unit = "Mt CO2 (2022)"
   )
+
+carbon_emissions <- carbon_emissions[-5,]
+carbon_emissions <- carbon_emissions[-5,]
+carbon_emissions <- carbon_emissions[-5,]
+carbon_emissions <- carbon_emissions[-5,]
+carbon_emissions <- carbon_emissions[-5,]
+carbon_emissions <- carbon_emissions[-5,]
+
+
+remaining_carbon_budgets <- carbon_emissions %>%
+  mutate(
+    remaining_carbon_budget = emissions_2022 * 4 + emissions_2030 * 4, # remaining carbon dbusget is the interpolized carbon budget until 2030 for a sector
```

Comment:
why do we multiply by 4? Please add a comment to explain what is going on there.

Reply:
@Antoine-Lalechere ?

Reply:
@Antoine-Lalechere do we include both 2022 and 2030 in the calculation? Or we start from 2023. Do I understand correctly then that we assume emissions assigned to years 2023-2026 are assumed to be the same as those in 2022 and those for 2027-2030 are assumed to be equal to 2030?

Reply:
We do an interpolation of carbon budget (so we have a line for carbon budget by sector between 2022 and 2030)
Then 2023 + 2029 carbon budget sums together and are equal to 2022 + 2030 carbon budget
same for 2024 + 2028 etc...
I think I forgot a year, it should be 4.5 and not 4 in the formula

Reply:
Ok, I think I understand. In the end after I run some calculations it seems that the sum of linearly interpolated points is the same as what you are doing here. The only unresolved question is if we should be using 2022 or not? Should we? Aren't we using data from 2023 to 2030 for the calculation?>

Reply:
The data we get is from 2022 so we have to consider it


Formula should actually be:

`remaining_carbon_budget = emissions_2022 * 4.5 + emissions_2030 * 4.5,`

This won't change the results as we take then percentage